src/str/basics.rs

```diff
@@ -288,3 +288,63 @@ where
         pattern: pattern.into(),
     }
 }
+
+/// Predicate that checks for all patterns.
+///
+/// This is created by `predicates::str:contains_all`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ContainsAllPredicate {
+    patterns: Vec<String>,
+}
+
+impl Predicate<str> for ContainsAllPredicate {
+    fn eval(&self, variable: &str) -> bool {
+        for pattern in &self.patterns {
+            if !variable.contains(pattern) {
+                return false;
+            }
+        }
+
+        true
+    }
```

Comment:
If we do this, we should probably have a `find_case` implementation to either report which items matched so people don't have to manually scan the string for them

Reply:
Okay, I will look into this!

Reply:
I added a basic `find_case` impl. I'm unsure if the current impl does what this function expects. This is because I'm not quite sure how it integrates into the `predicates-rs` ecosystem.

But, the implementation can be improved by implementing `reflection::PredicateReflection` and `fmt::Display` for `ContainsAllPredicate`.